src/pytorch_lightning/utilities/data.py

```diff
@@ -408,15 +410,15 @@ def _replace_dataloader_init_method() -> Generator[None, None, None]:
     """This context manager is used to add support for re-instantiation of custom (subclasses) of
     :class:`~torch.utils.data.DataLoader`. It patches the ``__init__`` method."""
     classes = _get_all_subclasses(DataLoader) | {DataLoader}
-    wrapped = set()
+    wrapped: set[Any] = set()
     for cls in classes:
-        if cls.__init__ not in wrapped:
+        if cls.__init__ not in wrapped and isinstance(cls, DataLoader):
             cls._old_init = cls.__init__
             cls.__init__ = _wrap_dataloader_init(cls.__init__)
             wrapped.add(cls.__init__)
     yield
     for cls in classes:
-        if hasattr(cls, "_old_init"):
+        if hasattr(cls, "_old_init") and isinstance(cls, DataLoader):
```

Comment:
This check is never gonna be true, since we are modifying classes and not instances - therefore `isinstance` call will always return False